src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -141,6 +135,20 @@ private void DualModeConnect_IPAddressToHost_Helper(IPAddress connectTo, IPAddre
                 Assert.True(socket.Connected);
             }
         }
+
+        private void DualModeConnect_IPAddressToHost_Fails_Helper(IPAddress connectTo, IPAddress listenOn)
+        {
+            Assert.ThrowsAny<SocketException>(() =>
+            {
+                DualModeConnect_IPAddressToHost_Helper(connectTo, listenOn, false);
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    // On Unix, socket assignment is random (not incremental) and there is a small chance the
+                    // listening socket was created in another test currently running. Try the test one more time.
+                    DualModeConnect_IPAddressToHost_Helper(connectTo, listenOn, false);
```

Comment:
I don't understand this.  If the first call to DualModeConnect_IPAddressToHost_Helper fails, won't it throw an exception?  In which case we'll never get to this line, which is meant to handle the case where it failed.